src/main/kotlin/de/itemis/mps/gradle/CreateDmg.kt

```diff
@@ -0,0 +1,85 @@
+package de.itemis.mps.gradle
+
+import org.gradle.api.GradleException
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.property
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FilterOutputStream
+import javax.inject.Inject
+
+open class CreateDmg @Inject constructor(
+        objects: ObjectFactory,
+        providers: ProviderFactory,
+        layout: ProjectLayout
+) : BaseBundleMacOsTask(objects, providers, layout) {
+    @InputFile
+    val backgroundImage = objects.fileProperty()
+
+    @OutputFile
+    val dmgFile = objects.fileProperty()
+            .convention(layout.buildDirectory.file("createDmg/$name/$name.dmg"))
```

Comment:
Shouldn’t the dmg file  not be placed under the dmg dir (L42) or do I missunderstand the usage of `dmgDir`?

Reply:
`dmgDir` existed previously, and it is a temporary directory for preparing the dmg contents.

`dmgFile` is an output `dmg` file which can be placed whenever you want.
Now I think it might make sense to use `buildDirectory.file("distributions/createDmg/$name/$name.dmg")` for the default name of the output file.

Reply:
However, it might be better to integrate with https://docs.gradle.org/current/userguide/distribution_plugin.html somehow.